### PR TITLE
Chore: move batch_size inside kind. BREAKING CHANGE.

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/core/model/kind.py [hunk 1]
```diff
@@ -108,18 +109,30 @@ def to_expression(self, dialect: str) -> exp.Column | exp.Tuple:
...
+    def _int_validator(cls, v: t.Any, field: ModelField) -> t.Optional[int]:
         if isinstance(v, exp.Expression):
             num = int(v.name)
         else:
             num = int(v)
         if num < 0:
+            raise ConfigError(
+                f"Invalid value {num} for {field.name}. The value should be greater than 0"
+            )
             raise ConfigError(f"Lookback must be an integer > 0")
```
Comment (vchan) [inline sqlmesh/core/model/kind.py:125]: This can be removed

File: sqlmesh/migrations/v0003_move_batch_size.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        model = snapshot["model"]
+        if "batch_size" in model:
+            batch_size = model.pop("batch_size")
+            kind = model.get("kind")
+
+            if kind:
+                if kind["name"] in ("INCREMENTAL_BY_TIME_RANGE", "INCREMENTAL_BY_UNIQUE_KEY"):
+                    kind["batch_size"] = batch_size
+
+                    # this is not efficient, i'm doing this because i'm lazy and no one has snapshots at the time of writing this migration
```
Comment (izeigerman) [inline sqlmesh/migrations/v0003_move_batch_size.py:21]: 😢 
image

File: sqlmesh/core/model/meta.py [hunk 3]
```diff
@@ -279,6 +256,11 @@ def lookback(self) -> int:
         """The incremental lookback window."""
         return (self.kind.lookback if isinstance(self.kind, _Incremental) else 0) or 0
 
+    @property
+    def batch_size(self) -> t.Optional[int]:
+        """The incremental lookback window."""
```
Comment (vchan) [inline sqlmesh/core/model/meta.py:261]: Is this right?

Comment (tobymao) [reply sqlmesh/core/model/meta.py]: nope thanks

**Review by vchan — APPROVED** [review]

**Pull request merged**